### PR TITLE
fix RDG highlights after code samples update

### LINKED ISSUE #32345: Fix formatting
### Description

[Enter feedback here]

### Page URL

https://learn.microsoft.com/en-us/aspnet/core/fundamentals/aot/request-delegate-generator/diagnostics/rdg011?view=aspnetcore-8.0&branch=pr-en-us-32344

### Content source URL

https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/fundamentals/aot/request-delegate-generator/diagnostics/rdg011.md

### Document ID

b6fd4969-c47b-cadd-c3d8-7d9deda76833

### Article author

@captainsafia

### LINKED ISSUE #32346: Fix snippet formatting
### Description

[Enter feedback here]
See https://github.com/dotnet/AspNetCore.Docs.Samples/pull/253

### Page URL

https://learn.microsoft.com/en-us/aspnet/core/fundamentals/aot/request-delegate-generator/diagnostics/rdg013?view=aspnetcore-8.0&branch=pr-en-us-32344

### Content source URL

https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/fundamentals/aot/request-delegate-generator/diagnostics/rdg013.md

### Document ID

646d1cae-5a1c-d315-6a8b-90d9630aff18

### Article author

@captainsafia

### LINKED ISSUE #32347: Fixes snippet formatting
### Description

🚨 Looks like you arrived from the internal review site 
🚨 Do not enter Microsoft confidential information here

[Enter feedback here]

### Page URL

https://learn.microsoft.com/en-us/aspnet/core/fundamentals/aot/request-delegate-generator/diagnostics/rdg010?view=aspnetcore-8.0&branch=pr-en-us-32344

### Content source URL

https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/fundamentals/aot/request-delegate-generator/diagnostics/rdg010.md

### Document ID

36843f44-e306-0fa5-77e2-6340dee034f6

### Article author

@captainsafia

### LINKED ISSUE #32348: Fix snippet formatting
### Description

See #32344

### Page URL

https://learn.microsoft.com/en-us/aspnet/core/fundamentals/aot/request-delegate-generator/diagnostics/rdg012?view=aspnetcore-8.0&branch=pr-en-us-32344

### Content source URL

https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/fundamentals/aot/request-delegate-generator/diagnostics/rdg012.md

### Document ID

98b9d6b6-fdd2-4600-2392-eb2c2cdd61f5

### Article author

@captainsafia

### PR DESCRIPTION
**Pull request by timdeschryver**
***EDIT by @Rick-Anderson*** Fixes #32345 Fixes #32346 Fixes #32347 Fixes #32348

@Rick-Anderson this PR updates the highlights after the samples were updated in https://github.com/dotnet/AspNetCore.Docs.Samples/pull/253

Side note: I noticed that the rdg001 page isn't included in the sidebar

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [aspnetcore/fundamentals/aot/request-delegate-generator/diagnostics/rdg010.md](https://github.com/dotnet/AspNetCore.Docs/blob/348ae2f16baf10916f7b902a5bbf0796547a281a/aspnetcore/fundamentals/aot/request-delegate-generator/diagnostics/rdg010.md) | [RDG010: InvalidAsParameters Nullable](https://review.learn.microsoft.com/en-us/aspnet/core/fundamentals/aot/request-delegate-generator/diagnostics/rdg010?branch=pr-en-us-32344) |
| [aspnetcore/fundamentals/aot/request-delegate-generator/diagnostics/rdg011.md](https://github.com/dotnet/AspNetCore.Docs/blob/348ae2f16baf10916f7b902a5bbf0796547a281a/aspnetcore/fundamentals/aot/request-delegate-generator/diagnostics/rdg011.md) | ["RDG011: Type parameters not supported"](https://review.learn.microsoft.com/en-us/aspnet/core/fundamentals/aot/request-delegate-generator/diagnostics/rdg011?branch=pr-en-us-32344) |
| [aspnetcore/fundamentals/aot/request-delegate-generator/diagnostics/rdg012.md](https://github.com/dotnet/AspNetCore.Docs/blob/348ae2f16baf10916f7b902a5bbf0796547a281a/aspnetcore/fundamentals/aot/request-delegate-generator/diagnostics/rdg012.md) | [RDG012: Unable to resolve inaccessible type](https://review.learn.microsoft.com/en-us/aspnet/core/fundamentals/aot/request-delegate-generator/diagnostics/rdg012?branch=pr-en-us-32344) |
| [aspnetcore/fundamentals/aot/request-delegate-generator/diagnostics/rdg013.md](https://github.com/dotnet/AspNetCore.Docs/blob/348ae2f16baf10916f7b902a5bbf0796547a281a/aspnetcore/fundamentals/aot/request-delegate-generator/diagnostics/rdg013.md) | [aspnetcore/fundamentals/aot/request-delegate-generator/diagnostics/rdg013](https://review.learn.microsoft.com/en-us/aspnet/core/fundamentals/aot/request-delegate-generator/diagnostics/rdg013?branch=pr-en-us-32344) |

### CODE REVIEW

File: aspnetcore/fundamentals/aot/request-delegate-generator/diagnostics/rdg010.md [hunk 1]
```diff
@@ -23,13 +23,13 @@ This diagnostic is emitted by the [Request Delegate Generator](/aspnet/core/fund
 
 The implementation of surrogate binding via the [`[AsParameters]`](xref:Microsoft.AspNetCore.Http.AsParametersAttribute) attribute in minimal APIs only supports types that are not nullable. 
 
-:::code language="csharp" source="~/../AspNetCore.Docs.Samples/fundamentals/aot/diagnostics/Rdg10/Program.cs" id="snippet_1" highlight="7":::
```
Comment (timdeschryver) [inline aspnetcore/fundamentals/aot/request-delegate-generator/diagnostics/rdg010.md:26]: image

File: aspnetcore/fundamentals/aot/request-delegate-generator/diagnostics/rdg011.md [hunk 2]
```diff
@@ -22,14 +22,13 @@ This diagnostic is emitted by the Request Delegate Generator when an endpoint co
 
 Endpoints that use generic type parameters are not supported. The endpoints within `MapEndpoints`  produce this diagnostic because of the generic `<T>` parameter.
 
-:::code language="csharp" source="~/../AspNetCore.Docs.Samples/fundamentals/aot/diagnostics/Rdg11/Program.cs" id="snippet_1" highlight="9-20":::
```
Comment (timdeschryver) [inline aspnetcore/fundamentals/aot/request-delegate-generator/diagnostics/rdg011.md:25]: image

File: aspnetcore/fundamentals/aot/request-delegate-generator/diagnostics/rdg011.md [hunk 3]
```diff
@@ -22,14 +22,13 @@ This diagnostic is emitted by the Request Delegate Generator when an endpoint co
...
+:::code language="csharp" source="~/../AspNetCore.Docs.Samples/fundamentals/aot/diagnostics/Rdg11/Program.cs" id="snippet_1" highlight="16":::
 
 ## How to fix violations
 
 Remove the generic type from endpoints.
 
-:::code language="csharp" source="~/../AspNetCore.Docs.Samples/fundamentals/aot/diagnostics/Rdg11/Program.cs" id="snippet_1f" highlight="9-20":::
```
Comment (timdeschryver) [inline aspnetcore/fundamentals/aot/request-delegate-generator/diagnostics/rdg011.md:31]: image

File: aspnetcore/fundamentals/aot/request-delegate-generator/diagnostics/rdg012.md [hunk 4]
```diff
@@ -23,14 +23,14 @@ This diagnostic is emitted by the Request Delegate Generator when an endpoint co
 Endpoints that use an inaccessible type (`private` or `protected`) are not supported. 
 The endpoints within `MapEndpoints` produce this diagnostic because of the `Todo` type has the `private` accessibility modifiers.
 
-:::code language="csharp" source="~/../AspNetCore.Docs.Samples/fundamentals/aot/diagnostics/Rdg12/Program.cs" id="snippet_1" highlight="9-20, 22":::
```
Comment (timdeschryver) [inline aspnetcore/fundamentals/aot/request-delegate-generator/diagnostics/rdg012.md:26]: image

File: aspnetcore/fundamentals/aot/request-delegate-generator/diagnostics/rdg012.md [hunk 5]
```diff
@@ -23,14 +23,14 @@ This diagnostic is emitted by the Request Delegate Generator when an endpoint co
...
+:::code language="csharp" source="~/../AspNetCore.Docs.Samples/fundamentals/aot/diagnostics/Rdg12/Program.cs" id="snippet_1" highlight="16-27, 29":::
 
 
 ## How to fix violations
 
 When applicable, set the target parameter type with a friendly accessibility.
 
-:::code language="csharp" source="~/../AspNetCore.Docs.Samples/fundamentals/aot/diagnostics/Rdg12/Program.cs" id="snippet_1f" highlight="9-20, 22":::
```
Comment (timdeschryver) [inline aspnetcore/fundamentals/aot/request-delegate-generator/diagnostics/rdg012.md:33]: image

File: aspnetcore/fundamentals/aot/request-delegate-generator/diagnostics/rdg013.md [hunk 6]
```diff
@@ -22,13 +22,14 @@ This diagnostic is emitted by the [Request Delegate Generator](/aspnet/core/fund
 
 ASP.NET Core supports resolving keyed and non-keyed services via [dependency injection](xref:fundamentals/dependency-injection). It's ***not*** feasible to resolve a service as both keyed and non-keyed. The following code  produces the diagnostic and throws a run time error with the same message:
 
-:::code language="csharp" source="~/../AspNetCore.Docs.Samples/fundamentals/aot/diagnostics/Rdg13/Program.cs" id="snippet_1" highlight="8":::
```
Comment (timdeschryver) [inline aspnetcore/fundamentals/aot/request-delegate-generator/diagnostics/rdg013.md:25]: image

File: aspnetcore/fundamentals/aot/request-delegate-generator/diagnostics/rdg013.md [hunk 7]
```diff
@@ -22,13 +22,14 @@ This diagnostic is emitted by the [Request Delegate Generator](/aspnet/core/fund
...
+:::code language="csharp" source="~/../AspNetCore.Docs.Samples/fundamentals/aot/diagnostics/Rdg13/Program.cs" id="snippet_1" highlight="14":::
 
 ## How to fix violations
 
 Resolve the target parameter as either a keyed or non-keyed service.
 
-:::code language="csharp" source="~/../AspNetCore.Docs.Samples/fundamentals/aot/diagnostics/Rdg13/Program.cs" id="snippet_1f" highlight="10,15,20":::
```
Comment (timdeschryver) [inline aspnetcore/fundamentals/aot/request-delegate-generator/diagnostics/rdg013.md:31]: image

**Review by timdeschryver — COMMENTED** [review]
Screenshots for the before (left) and after (right) states

**Review by Rick-Anderson — APPROVED** [review]

**Comment by Rick-Anderson** [comment]
> Side note: I noticed that the rdg001 page isn't included in the sidebar

[RDG001](https://learn.microsoft.com/en-us/aspnet/core/fundamentals/aot/request-delegate-generator/diagnostics/rdg001) is a stub held in reserve

This diagnostic is:

Reserved for a future version of ASP.NET Core.
Not currently emitted by the Request Delegate Generator.

**Pull request merged**